doc/specifications/drafts/fieldtype_generation/fieldtype_generation.md

```diff
@@ -0,0 +1,107 @@
+# FieldType generation proof-of-concept
+
+The file [`fieldtypes.yml`](fieldtypes.yml) describes the properties of a FieldType,
+the TextLine one, using yaml.
+
+It articulates around the definition of properties, for both the FieldType's Value
+and the FieldType's Definition (as in FieldDefinition, the "instance" of a FieldType
+in a ContentType).
+
+## Properties mapping
+
+Each property will be mapped to a Value property:
+
+The following configuration block:
+```yaml
+ezstring:
+    value_properties:
+        text:
+            type: string
+```
+
+Matches the following PHP Class:
+
+```php
+namespace eZ\Publish\Core\FieldType\TextLine;
+
+class Value implements ValueObject
+{
+    /**
+     * @var string
+     */
+    public $text;
+}
+```
+
+Properties can be defined in the same way for the field definition. However, due to how
+FieldTypes are architectured, it won't generate an object. Instead, it will generate
+code in the `validateValidatorConfiguration()` FieldType method.
+
+## Constraints mapping
+For each property, an optional constraints key can be defined:
+
+```yaml
+ezstring:
+    definition_properties:
+        minLength:
+            type: integer
+            constraints:
+                - Range: [0-255]
+                - LesserThanOrEqual: self.maxLength
+```
+
+The `minLength` FieldDefinition property has to be within the 0-255 range, and can't be
+higher than the maxLength property.
+
+The `self.<propertyName>` syntax is used to reference _other constraints_.
+
+A similar syntax can be used in value properties definitions:
+
+```yaml
+ezstring:
+    value_properties:
+        text:
+            type: string
+            constraints:
+                - Length:
+                    min: definition.minLength
```

Comment:
dont we miss here a way to tell eZ that we want to check min length? or is the 'Length' name of this contraint not just a label but in fact a class name / identifier of constraint type ?